--- SDLActivity.java.orginal	2020-12-21 17:44:36.000000000 +0000
+++ SDLActivity.java	2021-07-31 16:44:17.369775042 +0100
@@ -165,10 +165,17 @@
         return new String[] {
             "hidapi",
             "SDL2",
-            // "SDL2_image",
+            "SDL2_image",
             // "SDL2_mixer",
-            // "SDL2_net",
-            // "SDL2_ttf",
+            "SDL2_net",
+            "SDL2_ttf",
+            "cal3d",
+            "xml2",
+            "iconv",
+            "GLU",
+            "myglob",
+            "ssl",
+            "crypto",
             "main"
         };
     }
