--- SDLActivity.java.orginal	2024-11-13 20:17:04.299947657 +0000
+++ SDLActivity.java	2024-11-13 20:18:25.032265237 +0000
@@ -270,10 +270,15 @@
     protected String[] getLibraries() {
         return new String[] {
             "SDL2",
-            // "SDL2_image",
+            "SDL2_image",
             // "SDL2_mixer",
-            // "SDL2_net",
-            // "SDL2_ttf",
+            "SDL2_net",
+            "SDL2_ttf",
+            "cal3d",
+            "xml2",
+            "iconv",
+            "GLU",
+            "myglob",
             "main"
         };
     }
@@ -1936,6 +1941,7 @@
         ic = new SDLInputConnection(this, true);
 
         outAttrs.inputType = InputType.TYPE_CLASS_TEXT |
+                             InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD |
                              InputType.TYPE_TEXT_FLAG_MULTI_LINE;
         outAttrs.imeOptions = EditorInfo.IME_FLAG_NO_EXTRACT_UI |
                               EditorInfo.IME_FLAG_NO_FULLSCREEN /* API 11 */;
